src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -656,4 +702,41 @@ public static void exitGameIfFinished() {
     });
   }
 
+  /**
+   * todo, replace with something better
+   * Get the chat for the game, or null if there is no chat.
+   */
+  public static Chat getChat() {
+    final ISetupPanel model = setupPanelModel.getPanel();
+    if (model instanceof ServerSetupPanel) {
+      return model.getChatPanel().getChat();
+    } else if (model instanceof ClientSetupPanel) {
+      return model.getChatPanel().getChat();
+    } else {
+      return null;
+    }
+  }
+
+  public static boolean hasChat() {
+    return getChat() != null;
+  }
+
+  /**
+   * After the game has been left, call this.
+   */
+  public static void clientLeftGame() {
+    if (!SwingUtilities.isEventDispatchThread()) {
+      SwingAction.invokeAndWait(GameRunner::clientLeftGame);
+      return;
+    }
+    // having an oddball issue with the zip stream being closed while parsing to load default game. might be caused by
+    // closing of stream while unloading map resources.
+    ThreadUtil.sleep(100);
+    setupPanelModel.showSelectType();
+    showMainFrame();
+  }
+
+  public static void quitGame() {
+    mainFrame.dispatchEvent(new WindowEvent(mainFrame, WindowEvent.WINDOW_CLOSING));
```

Comment:
I'm not sure about adding something that is not necessary. It is debatable to add it here. Should we always opportunistically wrap any such code in an EDT guard? My default reaction is to say no and go with the (simpler) fewer lines of code.If this were something where there were a silent error, then I would be inclined otherwise (not the case here since we would get an error if we invoke this on the wrong thread).  So, I am on the fence on this one, I would rather not see us pepper almost any such code with EDT guards (but maybe that would be a good thing... debatable)